code-challenges.test.js

```diff
@@ -14,39 +14,130 @@
 
 // a) Create a test with an expect statement using the variable provided.
 
-const hitchhikersCharacters = [
-  { name: "ford prefect", occupation: "a hitchhiker" },
-  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+// describe("nameCapitolizer", () => {
+//   it("returns an array with a sentence about each person with their name capitalized.", () => {
+    const hitchhikersCharacters = [
+      { name: "ford prefect", occupation: "a hitchhiker" },
+      { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+      { name: "arthur dent", occupation: "a radio employee" }
+    ]
+    // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+//       expect(nameCapitolizer(hitchhikersCharacters).toEqual(
+//           ["Ford Prefect is a hitchhiker.", 
+//           "Zaphod Beeblebrox is president of the galaxy.", 
+//           "Arthur Dent is a radio employee."]))
+//   })
+// })
+
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
 
+// 1.) Declare the function to take in the array of objects.
+
+const nameCapitolizer = (people) => {
+
+// 2.) Create a for loop to iterate over the elements in the array.
+
+  const sentences = [];
+  for (let i = 0; i < people.length; i++) {
+
+// 3.) We iterate over the parameter(people[i]) and use the keys of the objects to access the value(name, occupation). 
+
+    const { name, occupation } = people[i];
+
+// 4.) We're going to split the name by each word (' ') and delclare it a new name(nameArray). Assign name array to the ) index.
+
+    let nameArray = name.split(' ')
+    nameArray[0]
+
+// 5.)The array is now split with the first name being the 0 index and last name being the 1 index. We'll use charAt to access the first letter of the names; toUpperCase to capitalize the strings; and the slice method at the 1 index to allow the toUppercase to only capialize the 0 index. This will be applied to the 0(first name) and the 1(last name) index.
+
+    const fullName = `${nameArray[0].charAt(0).toUpperCase()}${nameArray[0].slice(1)} ${nameArray[1].charAt(0).toUpperCase()}${nameArray[1].slice(1)}`;
+
+// Now we call upon our empty array and assign it to a string interpolation that has the fullName and occupation objects. Push that data into the empty array and return the array.
+
+    const sentence = `${fullName} is ${occupation}.`;
+    sentences.push(sentence);
+  }
+  return sentences;
+}
```

Comment:
There can definitely be some refactoring done to clean this up, but good logic overall!